demo/src/main/java/eu/kevin/demo/countryselection/adapters/CountryListAdapter.kt

```diff
@@ -0,0 +1,43 @@
+package eu.kevin.demo.countryselection.adapters
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import eu.kevin.common.architecture.BaseListAdapter
+import eu.kevin.common.extensions.getColorFromAttr
+import eu.kevin.common.extensions.setDebounceClickListener
+import eu.kevin.demo.R
+import eu.kevin.demo.countryselection.entities.Country
+import eu.kevin.demo.countryselection.helpers.CountryHelper
+import eu.kevin.demo.databinding.ItemCountryListBinding
+
+internal class CountryListAdapter(
+    override var items: List<Country> = emptyList(),
+    private val onCountryClicked: (String) -> Unit
+) : BaseListAdapter<Country, ItemCountryListBinding>(items) {
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
+        val inflater = LayoutInflater.from(parent.context)
+        return ViewHolder(ItemCountryListBinding.inflate(inflater, parent, false))
+    }
+
+    override fun onBindViewHolder(binding: ItemCountryListBinding, item: Country, position: Int) {
+        val context = binding.root.context
```

Comment:
having context as lateinit variable  in BaseListAdapter and initializing it in there would be a better option